### DISCUSSION TITLE
Wrong sorting behaviour

### DISCUSSION THREAD

**Discussion by MBach** [reupen/columns_ui#785] [Q&A] [answered]
I have a bug with the following settings:
Filter > Behaviour > sort tracks enabled:
`%album artist% - %date% - %album% - %discnumber% - %tracknumber% - %title%`
And with 
Playlist view > Grouping > (2 groups)
`%album artist%`
`'['%date%']' %album%`
![image](https://github.com/reupen/columns_ui/assets/1532657/e42cc76f-87ce-46e4-af56-baa0d5b24f2e)
Playlist view > Grouping > (3 groups)
![image](https://github.com/reupen/columns_ui/assets/1532657/dfce004e-f034-45fb-a92d-718bea9a411e)
It seems that lexicographical order takes precedence over discnumber and tracknumber

**Comment by reupen** [comment, marked as answer]
The Filters preferences page contains settings for Filter panels, but I don't see any in your screenshot. The playlist clearly isn't sorted by disc number. Try going to Edit / Sort / Sort by... and paste in your sorting string and see if the order is correct then.

If it is, you need to change the setting for whatever is sorting your playlist (perhaps Preferences / Shell Integration / Sort incoming files by).

**Reply by MBach** [reply]
You are right. It seems that when I've imported my playlist in newest Foobar 2.0, and started using Columns UI, the playlist wasn't sorted accordingly. I had to manually force the sort by selecting all and do the sort: sort by with my sort string.
